config/database.yml

```
@@ -1,25 +1,20 @@
-# SQLite version 3.x
-#   gem install sqlite3
-#
-#   Ensure the SQLite 3 gem is defined in your Gemfile
-#   gem 'sqlite3'
-#
 default: &default
-  adapter: sqlite3
+  adapter: postgresql
+  encoding: unicode
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
   timeout: 5000
 
```

Um alles zum laufen zu kriegen musste ich außerdem den Nutzer 'postgres' hinzufügen.
```suggestion
    username = postgres
```
Ob das sicherheits- (oder andersweitig) -technisch klug ist weiß ich aber ehrlich gesagt nicht. Immerhin könnte man ja auch bspw. für unsere Testumgebung und Dev-Umgebung und für Production jeweils Nutzer auf die Maschinen erzeugen und auf diese hier jeweils verweisen.

---

Sicher, dass das "username = postgres" und nicht "username: postgres"? So habe ich das gemacht, dann verschwand wenigstens der Fehler. Dennoch erhalte ich weiterhin Fehlermeldungen, genauer: "FATAL: database "db/sqrrl_development" does not exist".

---

Fehler wurde durch `rake db:create` behoben.
Danach musste ich ferner noch `rails db:migrate RAILS_ENV=development` ausführen.
Dies führte jedoch zu weiteren Fehlern

```== 20190530095000 ChangeRoleToIntegerMigration: migrating =====================
-- change_column(:users, :role, :integer, {:default=>0})
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::DatatypeMismatch: ERROR:  column "role" cannot be cast automatically to type integer
HINT:  You might need to specify "USING role::integer".
: ALTER TABLE "users" ALTER COLUMN "role" TYPE integer, ALTER COLUMN "role" SET DEFAULT 0
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `async_exec'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `block (2 levels) in execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:74:in `block in execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:581:in `block (2 levels) in log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:580:in `block in log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:571:in `log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:73:in `execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/schema_statements.rb:427:in `change_column'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:871:in `block in method_missing'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:840:in `block in say_with_time'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:840:in `say_with_time'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:860:in `method_missing'
/home/matthias/SWEP/db/migrate/20190530095000_change_role_to_integer_migration.rb:3:in `change'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:814:in `exec_migration'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:798:in `block (2 levels) in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:797:in `block in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:416:in `with_connection'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:796:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:977:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1292:in `block in execute_migration_in_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1343:in `block in ddl_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1343:in `ddl_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1262:in `each'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1210:in `block in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1210:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1036:in `up'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1011:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <main>'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/command.rb:48:in `invoke'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands.rb:18:in `<main>'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
bin/rails:4:in `<main>'

Caused by:
ActiveRecord::StatementInvalid: PG::DatatypeMismatch: ERROR:  column "role" cannot be cast automatically to type integer
HINT:  You might need to specify "USING role::integer".
: ALTER TABLE "users" ALTER COLUMN "role" TYPE integer, ALTER COLUMN "role" SET DEFAULT 0
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `async_exec'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `block (2 levels) in execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:74:in `block in execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:581:in `block (2 levels) in log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:580:in `block in log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:571:in `log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:73:in `execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/schema_statements.rb:427:in `change_column'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:871:in `block in method_missing'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:840:in `block in say_with_time'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:840:in `say_with_time'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:860:in `method_missing'
/home/matthias/SWEP/db/migrate/20190530095000_change_role_to_integer_migration.rb:3:in `change'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:814:in `exec_migration'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:798:in `block (2 levels) in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:797:in `block in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:416:in `with_connection'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:796:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:977:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1292:in `block in execute_migration_in_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1343:in `block in ddl_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1343:in `ddl_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1262:in `each'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1210:in `block in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1210:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1036:in `up'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1011:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <main>'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/command.rb:48:in `invoke'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands.rb:18:in `<main>'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
bin/rails:4:in `<main>'

Caused by:
PG::DatatypeMismatch: ERROR:  column "role" cannot be cast automatically to type integer
HINT:  You might need to specify "USING role::integer".
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `async_exec'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `block (2 levels) in execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:74:in `block in execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:581:in `block (2 levels) in log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:580:in `block in log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:571:in `log'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/database_statements.rb:73:in `execute'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql/schema_statements.rb:427:in `change_column'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:871:in `block in method_missing'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:840:in `block in say_with_time'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:840:in `say_with_time'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:860:in `method_missing'
/home/matthias/SWEP/db/migrate/20190530095000_change_role_to_integer_migration.rb:3:in `change'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:814:in `exec_migration'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:798:in `block (2 levels) in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:797:in `block in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:416:in `with_connection'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:796:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:977:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1292:in `block in execute_migration_in_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1343:in `block in ddl_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1343:in `ddl_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1262:in `each'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1210:in `block in migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1210:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1036:in `up'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/migration.rb:1011:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <main>'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/command.rb:48:in `invoke'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/railties-5.2.3/lib/rails/commands.rb:18:in `<main>'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/home/matthias/.rvm/gems/ruby-2.6.3/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
bin/rails:4:in `<main>'
Tasks: TOP => db:migrate
```

---

Ich habe nun oben meine Suggestion korrigiert, du hattest natürlich recht. Konnte man ja auch an den anderen Zeilen leicht sehen.

Zu deinen anderen Fehlermeldungen: Was passiert denn, wenn du mal rails db:migrate:reset und db:migrate ausführst? 
(und Sidenote: mal sollte anstatt rake lieber rails nutzen. Rake wurde noch vor Rails 5.0 verwendet, und jetzt hat man mit Rails 5.0 den Command rails.

---

Der Fehler wurde offenbar dadurch ausgelöst, dass bereits Datensätze in der Tabelle standen und ein Casting von String -> Int dadurch nicht möglich war.
Es konnte behoben werden durch `rake db:reset`